core/src/Streamly/Internal/Data/Stream/Container.hs

```diff
@@ -106,32 +109,33 @@ joinInner s1 s2 =
             Nothing -> Nothing
 
 -- XXX We can do this concurrently.
+-- XXX Check performance of StreamD vs StreamK
 -- XXX If the second stream is sorted and passed as an Array or a seek capable
 -- stream then we could use binary search if we have an Ord instance or
 -- Ordering returning function. The time complexity would then become (m x log
 -- n).
 
--- XXX Check performance of StreamD vs StreamK
-
--- | Like 'joinInner' but emit @(a, Just b)@, and additionally, for those @a@'s
--- that are not equal to any @b@ emit @(a, Nothing)@.
+-- | Like 'innerJoin' but emits @(a, Just b)@ whenever a and b are equal, for
+-- those @a@'s that are not equal to any @b@ emits @(a, Nothing)@.
 --
--- The second stream is evaluated multiple times. If the stream is a
--- consume-once stream then the caller should cache it in an 'Data.Array.Array'
--- before calling this function. Caching may also improve performance if the
--- stream is expensive to evaluate.
+-- This is a generalization of 'innerJoin' to include all elements from the
+-- left stream and not just those which have an equal in the right stream. This
+-- is not a commutative operation, the order of the stream arguments matters.
```

Comment:
```suggestion
-- is not a commutative operation, the order of the stream arguments matter.
```

Reply:
original is correct - `the order matters`.